### ISSUE TITLE
Modifying resource tags on a compute instance forces replacement

### ISSUE THREAD

**Issue by sbagdc** [hashicorp/terraform-provider-google#15148]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

Terraform v1.5.2

### Affected Resource(s)

* google_compute_instance

### Terraform Configuration Files

```tf
resource "google_compute_instance" "default" {
  name         = "instance-1"
  project      = "my-project-1234"
  machine_type = "e2-micro"
  zone         = "us-central1-a"

  boot_disk {
    initialize_params {
      image = "debian-cloud/debian-11"
    }
  }

  params {
    resource_manager_tags = {
      "tagKeys/1111" = "tagValues/1111"
      "tagKeys/2222" = "tagValues/2222" # Additional tag added to existing instance
    }
  }

  network_interface {
    subnetwork = "my/subnet/self/link"
  }
}
```

### Expected Behavior

When modifying the resource manager tags on a compute instance, the tags will be updated on the existing instance.

### Actual Behavior

Terraform wants to destroy and recreate the entire compute instance when adding, modifying, or deleting resource manager tags from the instance.

```tf
~ params {
    ~ resource_manager_tags = { # forces replacement
        + "tagKeys/2222" = "tagValues/2222"
          # (1 unchanged element hidden)
      }
  }
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Resource manager tags can be modified on an existing compute instance via the Google Cloud console and via the gcloud command. In both cases the tags are modified on the instance in place, without the need to destroy and recreate the instance. Not sure why Terraform forces replacement of the instance instead of modifying the tags in place like the other methods.

Thanks

**Comment by edwardmedia** [comment]
similar to https://github.com/hashicorp/terraform-provider-google/issues/12620

**Comment by rileykarson** [comment]
This is working as intended, although it is slightly suboptimal. I evaluated the best option for supporting this field alongside some members of the GCE team, and the current approach is what we landed on.

Basically- the GCE API can't return the current set of tags bound to the resource, and `resource_manager_tags` will overwrite all tags applied to the resource, even those applied through the resource manager API directly. To avoid Terraform silently clobbering tags without stating it would do so in plan, we elected to mark the field as immutable. This makes it effectively a creation-only field on the resource.
